packages/ui/src/app/components/PageHeaderWithHint.tsx

```diff
@@ -0,0 +1,61 @@
+import { AnimatePresence } from 'framer-motion'
+import React, { ReactNode, useCallback, useEffect } from 'react'
+
+import { PageHeaderRow, PageHeaderWrapper } from '@/app/components/PageLayout'
+import { ButtonsGroup } from '@/common/components/buttons'
+import { HintButton } from '@/common/components/buttons/HintButton'
+import { PageTitle } from '@/common/components/page/PageTitle'
+import { VideoHint } from '@/common/components/VideoHint'
+import { useLocalStorage } from '@/common/hooks/useLocalStorage'
+import { useToggle } from '@/common/hooks/useToggle'
+
+interface PageHeaderWithHintProps {
+  title: string
+  hintType: 'workingGroups' | 'proposals' | 'council'
+  buttons?: ReactNode
+  tabs?: ReactNode
+}
+
+const hintVideoURI: { [key in PageHeaderWithHintProps['hintType']]: string } = {
+  workingGroups: 'https://www.youtube.com/embed/cC-tTakpJEs',
+  proposals: 'https://www.youtube.com/embed/cC-tTakpJEs',
+  council: 'https://www.youtube.com/embed/cC-tTakpJEs',
+}
+
+export const PageHeaderWithHint = ({ title, hintType, buttons, tabs }: PageHeaderWithHintProps) => {
+  const [hintFinished, setHintFinished] = useLocalStorage<boolean>(hintType + '-hint')
+  const [experiencedCloseTooltip, setExperiencedCloseTooltip] = useLocalStorage<boolean>('experience-close-tooltip')
+  const [showCloseTooltip, toggleCloseTooltip] = useToggle(false)
+  const [isHintVisible, toggleHint] = useToggle(!hintFinished)
+
+  useEffect(() => {
+    if (!isHintVisible && !hintFinished) {
+      setHintFinished(true)
+    }
+  }, [isHintVisible])
+
+  const closeHint = useCallback(() => {
+    toggleHint()
+
+    if (!experiencedCloseTooltip) {
+      toggleCloseTooltip()
+      setExperiencedCloseTooltip(true)
+    }
+  }, [])
```

Comment:
I see 4 booleans here that control whether to display the hint video or not. I might be wrong here, but let's take another look at those. Could you write shortly what's the difference between two local storage stored values?

AFAICS, the states for this component are:

*   new user (not closed ever)
*   opened
*   closed

Am I missing something here?

Reply:
So, **hintFinished** is used to control _not closed ever_.
And **experiencedCloseTooltip** is used to control _the closing tooltip_ - I assume that we should show it only for first time closing on cross icon
![image](https://user-images.githubusercontent.com/83761318/140718727-aaacfe9b-38ea-46b0-b941-32838e94ff39.png)

Reply:
OH, OK. I didn't see that while testing. That makes sense then.

Reply:
OH, OK. I didn't see that while testing. That makes sense then.

Reply:
@DevNikita16 but please check the logic of that anyway. It seems that this doesn't work as expected.

I cleared local storage, closed the video using 'x'  and i did not saw the tooltip.

Reply:
@jodator © _The only thing left is close tooltip helper. @ClumsyVlad will handle it next Friday (#1732), I've prepared handling state for this tooltip_